src/ast/mod.rs

```diff
@@ -1289,12 +1267,19 @@ impl fmt::Display for Expr {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             Expr::Identifier(s) => write!(f, "{s}"),
-            Expr::MapAccess { column, keys } => {
-                write!(f, "{column}{}", display_separated(keys, ""))
-            }
             Expr::Wildcard(_) => f.write_str("*"),
             Expr::QualifiedWildcard(prefix, _) => write!(f, "{}.*", prefix),
             Expr::CompoundIdentifier(s) => write!(f, "{}", display_separated(s, ".")),
+            Expr::CompoundExpr { root, chain } => {
+                write!(f, "{}", root)?;
+                for field in chain {
+                    match field {
+                        AccessField::Expr(expr) => write!(f, ".{}", expr)?,
```

Comment:
I would have expected the `,` or the `[]` be in the `AccessField` so it can be self-displaying

Reply:
Indeed. It's better.